lib/go-tc/servers.go

```diff
@@ -495,6 +745,48 @@ type CommonServerProperties struct {
 	XMPPPasswd       *string              `json:"xmppPasswd" db:"xmpp_passwd"`
 }
 
+// CommonServerPropertiesV40 is just the collection of properties which are
+// shared by all servers across API versions >= 4.0.
+type CommonServerPropertiesV40 struct {
```

Comment:
I don't think this needs to be separate from the ServerV4 definition; instead I'd just remove the router fields from `CommonServerProperties` and add them to the older objects that still need them. The point of `CommonServerProperties` is just to not re-iterate properties shared by servers of all versions - but people are supposed to access those properties like `server.Prop` not `server.CommonServerProperties.Prop` so it's fine to change it as long as the overall Server structure isn't changed.